### PR TITLE
feat: run under a subpath

### PR DESCRIPTION
**Pull request by eh-am**
Adds a `-api.base-url` flag to support running the UI under a different baseUrl.
Closes https://github.com/grafana/phlare/issues/638

# How it works
When doing a prod build (`make build`), it generates an `index.html` with `<base href={{ .BaseURL }} />` that's parsed as a go template by the phlare server and injects the flag value (with some adaptations).

It accepts absolute URLs (such as `http://localhost:8080/foobar/`) and paths `(/foobar/`).

There's some dirtiness like stripping `/ui/` when doing requests to the backend, which hopefully would be cleaned up once we move the ui to the root.

Also adds e2e tests using cypress to validate this works.

- [X] Add e2e tests
- [ ] Normalize terminology between "baseURL", "basePath", "baseName" etc

### CODE REVIEW

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18.15.0
+          cache: yarn
+      - name: Install Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.19.8
```
Comment (simonswine) [inline .github/workflows/e2e.yaml:28]: To match the other pipelines

```suggestion
          go-version: "1.19.9"
```

File: .github/workflows/e2e.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
```
Comment (simonswine) [inline .github/workflows/e2e.yaml:57]: To match the other pipelines

```suggestion
          go-version: "1.19.9"
```

**Review by simonswine — APPROVED** [review]
LGTM.

Also great to get some e2e test coverage

File: .github/workflows/e2e.yaml (hunk 2 above)
Comment (eh-am) [reply .github/workflows/e2e.yaml:57]: Half of the work is just building the ui + binary :( Not sure what can be done there

Comment (simonswine) [reply .github/workflows/e2e.yaml:57]: I think the `cache: true` might already shave off quite a bit

**Pull request merged**